pkg/rpc/client/rpc.go

```diff
@@ -545,3 +558,17 @@ func (c *Client) ValidateAddress(address string) error {
 	}
 	return nil
 }
+
+// CalculateValidUntilBlock calculates ValidUntilBlock field for tx.
+func (c *Client) CalculateValidUntilBlock() (uint32, error) {
+	var result uint32
+	blockCount, err := c.GetBlockCount()
+	if err != nil {
+		return result, errors.Wrapf(err, "cannot get block count")
+	}
+	validators, err := c.GetValidators()
```

Comment:
I fear this is quite an expensive call, doing it often may be a bit problematic, so it either should be cached (for some block number) or substituted with `len(standbyvalidators)`.

Reply:
However, we don't have `standbyvalidators` here as there's no such RPC-call

Reply:
I'd thought about config file, but I guess for RPC client even that might be unavailable, so let's cache this call (its results don't change very often) for some 100 blocks (a little less than half an hour given the default block time setting).

Reply:
Implemented cache like field in Client struct (see: https://github.com/nspcc-dev/neo-go/pull/856/commits/e36cd5a4d7af8fca01d23252c6072f57778a9250), but not sure that it's a good solution as every time we create new client and call GetValidators, we'll still re-send "getvalidators" request. 

I also thought about [httputil.DumpResponce(...)](https://golang.org/pkg/net/http/httputil/#DumpResponse) method, but here we have to save the dump somewhere (on disk?)

Reply:
> every time we create new client and call GetValidators, we'll still re-send "getvalidators" request.

It should be OK, think about some long-term service that operates using our client (like gorshok!), it will generate one transaction after another using the same client instance and it will benefit from this caching.

At the same time, caching whole `GetValidators` response is not something we want, it has a direct mapping to the RPC call, so when I call `GetValidators` I expect to see actual server response every time. But it's OK to define whatever semantics we want for auxiliary higher-level methods like `CalculateValidUntilBlock` (as long as this semantics is correctly explained to the user of this method). So caching should be done in this function and only for validators count (we don't care about validators list itself here).

We may change it in the future when we're to have event subscription.

Reply:
Fixed this and added annotation for caching structures in Client.